### ISSUE TITLE
Add a setting to make all workspace projects "injected"

### ISSUE THREAD

**Issue by zkochan** [pnpm/pnpm#8332]
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

Injected dependencies are currently hard to use because you need to specify it separately for every single workspace dependency

### Describe the solution you'd like

Instead of setting "injected: true" for every workspace project in dependencies, have a single setting in `.npmrc` that may be set to `true`.

### Describe the drawbacks of your solution

No drawbacks

### Describe alternatives you've considered

No alternatives.

**Comment by runspired** [comment]
this has been added; however, its currently not usable for any realistic project because in order to make use of hardlinks you have to regularly re-sync them. This may happen because of a build that occurs in multiple stages (E.g. packages built first are consumed by packages built later, in which case a sync of the hardlinks must occur in between) or because active dev in the project has changed the files or code since the last run of the install command.

The best tool for handling re-sync of hardlinks in pnpm workspaces is [pnpm-sync-dependencies-meta-injected](https://github.com/NullVoxPopuli/pnpm-sync-dependencies-meta-injected/tree/main), however it currently only supports the more verbose syntax. I've opened an [issue](https://github.com/NullVoxPopuli/pnpm-sync-dependencies-meta-injected/issues/7) to track changing this there, but to be honest it feels like it would be better for pnpm to expose a utility to regenerate the hardlinks

**Comment by zkochan** [comment]
We have also added an option for syncing: [sync-injected-deps-after-scripts](https://pnpm.io/npmrc#sync-injected-deps-after-scripts).

**Comment by runspired** [comment]
@zkochan thanks for closing and following up! I'd later discovered this myself, works great!

**Comment by gbleu** [comment]
Would it make sense to also have a `sync-injected-deps-before-scripts` ?

In a distributed system, the actual build downloads the dependencies on a new node where the hardlinks are not refreshed.

**Comment by runspired** [comment]
@gbleu yeah I hit the need for this with turborepo immediately. I have to deactivate the turbo cache for initial builds because the scripts are never run and so pnpm never refreshes the hardlinks for the assets restored from cache.

I ended up doing a thing that
